api/handler/multipart_upload.go

```diff
@@ -0,0 +1,536 @@
+		XMLName  xml.Name `xml:"http://s3.amazonaws.com/doc/2006-03-01/ InitiateMultipartUploadResult" json:"-"`
+		Bucket   string   `xml:"Bucket"`
+		Key      string   `xml:"Key"`
+		UploadID string   `xml:"UploadId"`
+	}
+
+	CompleteMultipartUploadResponse struct {
+		XMLName xml.Name `xml:"http://s3.amazonaws.com/doc/2006-03-01/ CompleteMultipartUploadResult" json:"-"`
+		Bucket  string   `xml:"Bucket"`
+		Key     string   `xml:"Key"`
+		ETag    string   `xml:"ETag"`
+	}
+
+	ListMultipartUploadsResponse struct {
+		XMLName            xml.Name          `xml:"http://s3.amazonaws.com/doc/2006-03-01/ ListMultipartUploadsResult" json:"-"`
+		Bucket             string            `xml:"Bucket"`
+		CommonPrefixes     []CommonPrefix    `xml:"CommonPrefixes"`
+		Delimiter          string            `xml:"Delimiter,omitempty"`
+		EncodingType       string            `xml:"EncodingType,omitempty"`
+		IsTruncated        bool              `xml:"IsTruncated"`
+		KeyMarker          string            `xml:"KeyMarker"`
+		MaxUploads         int               `xml:"MaxUploads"`
+		NextKetMarker      string            `xml:"NextKetMarker,omitempty"`
+		NextUploadIdMarker string            `xml:"NextUploadIdMarker,omitempty"`
+		Prefix             string            `xml:"Prefix"`
+		Uploads            []MultipartUpload `xml:"Upload"`
+		UploadIdMarker     string            `xml:"UploadIdMarker,omitempty"`
+	}
+
+	ListPartsResponse struct {
+		XMLName              xml.Name      `xml:"http://s3.amazonaws.com/doc/2006-03-01/ ListPartsresult" json:"-"`
+		Bucket               string        `xml:"Bucket"`
+		Initiator            Initiator     `xml:"Initiator"`
+		IsTruncated          bool          `xml:"IsTruncated"`
+		Key                  string        `xml:"Key"`
+		MaxParts             int           `xml:"MaxParts,omitempty"`
+		NextpartNumberMarker int           `xml:"NextpartNumberMarker,omitempty"`
+		Owner                Owner         `xml:"Owner"`
+		Parts                []*layer.Part `xml:"Part"`
+		PartNumberMarker     int           `xml:"PartNumberMarker,omitempty"`
+		StorageClass         string        `xml:"StorageClass,omitempty"`
+		UploadId             string        `xml:"UploadId"`
+	}
+
+	MultipartUpload struct {
+		Initiated    string    `xml:"Initiated"`
+		Initiator    Initiator `xml:"Initiator"`
+		Key          string    `xml:"Key"`
+		Owner        Owner     `xml:"Owner"`
+		StorageClass string    `xml:"StorageClass,omitempty"`
+		UploadID     string    `xml:"UploadId"`
+	}
+
+	Initiator struct {
+		ID          string `xml:"ID"`
+		DisplayName string `xml:"DisplayName"`
+	}
+
+	CompleteMultipartUpload struct {
+		XMLName xml.Name               `xml:"http://s3.amazonaws.com/doc/2006-03-01/ CompleteMultipartUpload"`
+		Parts   []*layer.CompletedPart `xml:"Part"`
+	}
+)
+
+const (
+	uploadIdHeaderName   = "uploadId"
+	partNumberHeaderName = "partNumber"
+	maxSizeUploadsList   = 1000
+	maxSizePartsList     = 1000
```

Comment:
Maybe we should define all s3 related constants in one place? For example all list related operations have the same default max items value 1000. At least I would prefer to hold [multipart upload limits](https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html) together